sync.go

```diff
@@ -201,13 +195,12 @@ func (lw *LibWallet) SpvSync(peerAddresses string) error {
 	syncer := spv.NewSyncer(loadedWallet, lp)
 	syncer.SetNotifications(lw.spvSyncNotificationCallbacks())
 	if len(validPeerAddresses) > 0 {
-		syncer.SetPersistantPeers(validPeerAddresses)
+		syncer.SetPersistentPeers(validPeerAddresses)
 	}
 
 	loadedWallet.SetNetworkBackend(syncer)
-	lw.walletLoader.SetNetworkBackend(syncer)
 
-	ctx, cancel := lw.contextWithShutdownCancel(context.Background())
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
With this approach, sync won't get canceled when an interrupt signal is received.
To fix, also execute `lw.CancelSync()` (if `lw.CancelSync != nil` of course) after canceling all contexts created with `lw.shutdownContext()`. This should be done when the `lw.shuttingDown` signal is received in the go routine in `lw.listenForShutdown()`.

This means that there would be no need to do `lw.CancelSync()` in `lw.Shutdown()` because `lw.Shutdown()` sends a signal to `lw.shuttingDown` to cancel all contexts created with `lw.shutdownContext()`.